share/eds/eds.go

```diff
@@ -45,28 +46,35 @@ type writingSession struct {
 // Order: [ Carv1Header | Q1 |  Q2 | Q3 | Q4 | inner nodes ]
 // For more information about the header: https://ipld.io/specs/transport/car/carv1/#header
 func WriteEDS(ctx context.Context, eds *rsmt2d.ExtendedDataSquare, w io.Writer) error {
+	ctx, span := tracer.Start(ctx, "write-eds")
+	defer span.End()
+
 	// 1. Reimport EDS. This is needed to traverse the NMT tree and cache the inner nodes (proofs)
 	writer, err := initializeWriter(ctx, eds, w)
 	if err != nil {
+		span.SetStatus(codes.Error, err.Error())
```

Comment:
is `span.RecordError` not required to be called in here? Check #1589

Reply:
It's not